ow_core/text_search_manager.php

```diff
@@ -0,0 +1,250 @@
+    }
+
+    /**
+     * Constructor.
+     */
+    private function __construct()
+    {}
+
+    /**
+     * Add entity
+     *
+     * @param string $type
+     * @param integer $id
+     * @param string  $searchText
+     * @param array $tags
+     * @param boolean $isActive
+     * @return boolean
+     */
+    public function addEntity( $type, $id, $searchText, array $tags = array(), $isActive = true )
+    {
+        $result =  self::getDefaultStorageInstance()->addEntity($type, $id, $searchText, $tags, $isActive);
+
+        if ( $result && self::getActiveStorageInstance() )
+        {
+            $result =  self::getActiveStorageInstance()->addEntity($type, $id, $searchText, $tags, $isActive);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Delete entity
+     *
+     * @param string $type
+     * @param integer $id
+     * @return boolean
+     */
+    public function deleteEntity( $type, $id )
+    {
+        $result =  self::getDefaultStorageInstance()->deleteEntity($type, $id);
+
+        if ( $result && self::getActiveStorageInstance() )
+        {
+            $result =  self::getActiveStorageInstance()->deleteEntity($type, $id);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Delete all entities
+     *
+     * @param string $type
+     * @return boolean
+     */
+    public function deleteAllEntities( $type = null )
+    {
+        $result =  self::getDefaultStorageInstance()->deleteAllEntities($type);
+
+        if ( $result && self::getActiveStorageInstance() )
+        {
+            $result =  self::getActiveStorageInstance()->deleteAllEntities($type);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Deactivate all entities
+     *
+     * @param string $type
+     * @return boolean
+     */
+    public function deactivateAllEntities( $type = null )
+    {
+        $result = self::getDefaultStorageInstance()->deactivateAllEntities($type);
+
+        if ( $result && self::getActiveStorageInstance() )
+        {
+            $result =  self::getActiveStorageInstance()->deactivateAllEntities($type);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Activate all entities
+     *
+     * @param string $type
+     * @return boolean
+     */
+    public function activateAllEntities( $type = null )
+    {
+        $result = self::getDefaultStorageInstance()->activateAllEntities($type);
+
+        if ( $result && self::getActiveStorageInstance() )
+        {
+            $result =  self::getActiveStorageInstance()->activateAllEntities($type);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Search entities
+     *
+     * @param string $searchText
+     * @param integer $first
+     * @param integer $limit
+     * @param array $tags
+     * @param boolean $sortByDate - sort by date or by relevance
+     * @return array
+     */
+    public function searchEntities( $searchText, $first, $limit, array $tags = array(), $sortByDate = false )

```

Comment:
We decided don't change this behaviour  because it very hard use inner search storage constants. Also it not needed now